synapse/types/__init__.py

```diff
@@ -592,7 +658,9 @@ def copy_and_advance(self, other: "RoomStreamToken") -> "RoomStreamToken":
             for instance in set(self.instance_map).union(other.instance_map)
         }
 
-        return RoomStreamToken(None, max_stream, immutabledict(instance_map))
+        return RoomStreamToken(
+            stream=max_stream, instance_map=immutabledict(instance_map)
+        )
```

Comment:
I think most of this is `super().copy_and_advance(...)`?

Reply:
whoops, that was what I was planning on doing